TypeCobol/Compiler/CompilationDocument.cs

```diff
@@ -284,118 +284,115 @@ public void UpdateTextLines(TextChangedEvent textChangedEvent)
-                                continue;
-                            if (compilationDocumentLines.Count <= textChange.LineIndex) //Avoid line remove exception
-                                continue;
+                    // Check if the last line was deleted
+                    int lastLineIndex = update.LineEnd;
+                    if (update.LineEnd > update.LineStart && update.ColumnEnd == 0)
+                    {
+                        //Allows to detect if the next line was suppressed
+                    }
+                    if (update.Text?.Length == 0)
+                    {
+                        lineUpdates = new List<string>();
+                    }
 
-                            compilationDocumentLines.RemoveAt(textChange.LineIndex);
-                            encounteredCodeElement = false; //Will allow to update allow line index without erasing all diagnostics after the first encountered line with CodeElements
+                    if (lastLineIndex > firstLineIndex)
+                    {
+                        originalLastLineText = compilationDocumentLines[Math.Min(lastLineIndex, compilationDocumentLines.Count - 1)].Text;
+                    }
 
-                            foreach (var lineToUpdate in compilationDocumentLines.Skip(textChange.LineIndex)) //Loop on every line that appears after deleted line
-                            {
-                                //Remove generated diagnostics for the line below the deleted line.
-                                if (!encounteredCodeElement)
-                                    lineToUpdate.ResetDiagnostics();
+                    // Text not modified at the beginning of the first replaced line
+                    string startOfFirstLine = null;
+                    if (firstLineChar > 0)
+                    {
+                        if (originalFirstLineText.Length >= update.ColumnStart)
+                            startOfFirstLine = originalFirstLineText.Substring(0, update.ColumnStart);
+                        else
+                            startOfFirstLine = originalFirstLineText.Substring(0, originalFirstLineText.Length)
+                                             + new string(' ', update.ColumnStart - originalFirstLineText.Length);
+                    }
 
-                                lineToUpdate.ShiftUp();
+                    // Text not modified at the end of the last replaced line
+                    string endOfLastLine = null;
+                    if (update.ColumnEnd < originalLastLineText.Length)
+                    {
+                        endOfLastLine = originalLastLineText.Substring(update.ColumnEnd);
+                    }
 
-                                if (lineToUpdate.CodeElements != null)
-                                    encounteredCodeElement = true; 
-                            }
+                    // Remove all the old lines
```

Comment:
I think we should try to merge the removal and insertion of lines in order to use `TextChangeType.LineUpdated`.
The goal is to optimize by limiting the update of LineIndex of all lines after the insertion of removal.

Reply:
I've created issue #2416 for that.